staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -175,10 +175,31 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 	options := metav1.ListOptions{ResourceVersion: "0"}
 	r.metrics.numberOfLists.Inc()
 	start := r.clock.Now()
-	list, err := r.listerWatcher.List(options)
+
+	var list runtime.Object
+	var err error
+	listCh := make(chan struct{}, 1)
```

Comment:
make it a `chan err` and get rid of the shared variable, same for panic.

Reply:
it returns runtime.object as well making you wrap stuff in structs for the tuple which didn't seem to worth it